techniques/system/rudderWebapp/1.0/integrity-check.cf

```diff
@@ -0,0 +1,64 @@
+####################################################################################
+# Copyright 2021 Normation SAS
+####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+bundle agent system_root_integrity_check(trackingkey, technique_name)
+{
+  vars:
+      "key_destination" string => "${sys.workdir}/ppkeys/root-${sys.key_digest}.pub";
+      "component"       string => "Check configuration-repository";
+      "component_key1"  string => "Configuration-repository exists";
+      "component_key2"  string => "Check configuration-repository GIT lock";
+
+  classes:
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  files:
+      "${g.rudder_configuration_repository}"
+        create => "true",
+        action => WarnOnly,
+        classes => if_else("rudder_integrity_ok", "rudder_integrity_failed");
+
+      "${g.rudder_configuration_repository}/.git/index.lock"
```

Comment:
What happens if we have a slow git command running ?

Reply:
I believe that we should avoid when possible to rely on data taken from other resources.

Reply:
Wrong comment, I have no idea, I just took the existing techniques and port them.